diff --git a/src/main.c b/src/main.c
index 24157b18c..d2570a24f 100644
--- a/src/main.c
+++ b/src/main.c
@@ -52,6 +52,7 @@ static struct {
 	gboolean print_config;
 	gboolean become_daemon;
 	gboolean g_fatal_warnings;
+	gboolean clr_all_ifs; /* in Clear Linux: is capable of managing all the interfaces (previously patched to ignore wired). */
 	gboolean run_from_build_dir;
 	char *opt_log_level;
 	char *opt_log_domains;
@@ -183,6 +184,7 @@ do_early_setup (int *argc, char **argv[], NMConfigCmdLineOptions *config_cli)
 		{ "pid-file", 'p', 0, G_OPTION_ARG_FILENAME, &global_opt.pidfile, N_("Specify the location of a PID file"), NM_DEFAULT_PID_FILE },
 		{ "run-from-build-dir", 0, 0, G_OPTION_ARG_NONE, &global_opt.run_from_build_dir, "Run from build directory", NULL },
 		{ "print-config", 0, 0, G_OPTION_ARG_NONE, &global_opt.print_config, N_("Print NetworkManager configuration and exit"), NULL },
+		{ "clr-all-ifs", 0, 0, G_OPTION_ARG_NONE, &global_opt.clr_all_ifs, N_("Internal Clear Linux use"), NULL },
 		{NULL}
 	};
 
@@ -263,6 +265,10 @@ main (int argc, char *argv[])
 
 	do_early_setup (&argc, &argv, config_cli);
 
+	if (global_opt.clr_all_ifs) {
+		exit (0);
+	}
+
 	if (global_opt.g_fatal_warnings)
 		_set_g_fatal_warnings ();
 
